app/components/autocomplete/channel_mention_item/index.ts

```diff
@@ -7,19 +7,24 @@ import {General} from '@mm-redux/constants';
 import {getChannel} from '@mm-redux/selectors/entities/channels';
 import {getTheme} from '@mm-redux/selectors/entities/preferences';
 import {getUser} from '@mm-redux/selectors/entities/users';
+import {GlobalState} from '@mm-redux/types/store';
 import {getChannelNameForSearchAutocomplete} from '@selectors/channel';
 import {isGuest as isGuestUser} from '@utils/users';
 
 import ChannelMentionItem from './channel_mention_item';
 
-function mapStateToProps(state, ownProps) {
+type OwnProps = {
+    channelId: string;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     const channel = getChannel(state, ownProps.channelId);
     let displayName = getChannelNameForSearchAutocomplete(state, ownProps.channelId);
 
     let isBot = false;
     let isGuest = false;
     if (channel?.type === General.DM_CHANNEL) {
-        const teammate = getUser(state, channel.teammate_id);
+        const teammate = channel.teammate_id ? getUser(state, channel.teammate_id) : undefined;
```

Comment:
why this change? did you find a case where teammate_id is not present for DM channels ?

Reply:
No, but I am not a fan of using the `!` operator. Nevertheless, we can suppose teammate_id is always defined, and use it.